toolkits/global/packages/global-forms/README.md

```diff
@@ -13,3 +15,265 @@ The `global-forms` component currently uses the `DEFAULT` brand only.
+2. Organize your fields into fieldsets using the data structure exemplified in `demo/context.json`. In this case, you will have to iterate over a `fieldsets` property something like this:
+
+```
+<form action="some/url">
+    {{#with myFormData}}
+		{{#with errorSummary}}
+			{{> errorSummary }}
+		{{/with}}
+		{{#each fieldsets}}
+			{{> fieldset }}
+		{{/each}}
+    {{/with}}
+</form>
+```
+
+### Fieldsets
+
+Fieldsets are used to group fields itemized under their `fields` property. If you do not want to include a (visible; screen reader identifiable) fieldset element or `<legend>`, simply omit the `legend` property. The following example represents a simple form body with a single, unlabeled fieldset containing two text inputs:
+
+```
+"fieldsets": {
+    [
+        {
+            "fields": [
+                {
+                    "type": "text",
+                    "label": "Your name",
+                    "id": "your-name",
+                    "name": "your-name"
+                },
+                {
+                    "type": "text",
+                    "label": "Your email",
+                    "id": "your-email",
+                    "name": "your-email"
+                }
+            ]
+        }
+    ]
+}
+```
+
+Where you do wish to include a legend, HTML is permissable, meaning you can include a heading to reinforce the form and page structure:
+
+```
+"fieldsets": {
+    [
+        {
+            "legend": "<h2>My level 2 legend</h2>",
+            "fields": [
+                ...
+            ]
+        }
+    ]
+}
+```
+
+(**Note:** it is valid and conforming to use heading elements inside `<legend>`s.)
+
+### Fields
+
+Fields take the `type` property to determine which of the `view/fields` they represent. This property along with `label`, `id`, and `name` should be mandatory in terms of your model.
+
+A wide range of standard input/field attributes are supported. So, for example, if you wanted to include a `readonly` attribute on your text input, you would just include a property of the same name on the data:
+
+```
+{
+    "type": "text",
+    "label": "Your email",
+    "id": "your-email",
+    "name": "your-email",
+    "readonly": true
+}
+```
+
+The `hint` property includes supplementary text under the main `<label>` text but _inside_ the `<label>`, meaning it is automatically available to screen reader software.
+
+```
+"hint": "This will be the email you signed up using"
+```
+
+The `optional` property affects the field’s label, appending _“(optional)”_ to the label text.
+
+```
+"optional": true
+```
+
+### Errors
+
+Each field can have an `error` property. The presence of the property indicates the field is in an error state and the property value (a string) defines the error message.
+
+```
+{
+    "type": "text",
+    "label": "Your name",
+    "id": "your-name",
+    "name": "your-name",
+    "error": "That is not a name, it’s a number"
```

Comment:
more of a 'stub'. Does SN Digital have defined microcopy for things like form errors? I wonder if a) it's worth putting anywhere here (probably not) or b) making sure it's part of the Elements Design System site for folks to reference.